src/PockyBot.NET/Services/Triggers/Unpeg.cs

```diff
@@ -0,0 +1,309 @@
+    {
+        private readonly IRandomnessHandler _randomnessHandler;
+        private readonly IChatHelper _chatHelper;
+        private readonly IBackgroundTaskQueue _backgroundTaskQueue;
+        private readonly IAsyncDelayer _asyncDelayer;
+        private readonly ILogger<Unpeg> _logger;
+        private readonly string[] _separatorWords = { "for", "because" };
+
+        public string Command => Commands.Unpeg;
+
+        public bool DirectMessageAllowed => false;
+
+        public bool CanHaveArgs => true;
+
+        public Role[] Permissions => Array.Empty<Role>();
+
+        public Unpeg(IRandomnessHandler randomnessHandler, IChatHelper chatHelper, IBackgroundTaskQueue backgroundTaskQueue, IAsyncDelayer asyncDelayer, ILogger<Unpeg> logger)
+        {
+            _randomnessHandler = randomnessHandler;
+            _chatHelper = chatHelper;
+            _backgroundTaskQueue = backgroundTaskQueue;
+            _asyncDelayer = asyncDelayer;
+            _logger = logger;
+        }
+
+        public Task<Message> Respond(Message message)
+        {
+            var recipientIsPerson = true;
+            var recipientName = message.MessageParts[1].Text.Trim().Substring(Command.Length).Trim();
+            string recipientId = null;
+
+            if (string.IsNullOrEmpty(recipientName))
+            {
+                if (message.MessageParts.Length < 3 || message.MessageParts[2].MessageType != MessageType.PersonMention)
+                {
+                    return Task.FromResult(new Message
+                    {
+                        Text = "Please provide someone or something to unpeg."
+                    });
+                }
+
+                recipientName = message.MessageParts[2].Text;
+                recipientId = message.MessageParts[2].UserId;
+            }
+            else
+            {
+                recipientIsPerson = false;
+                recipientName = GetRecipientNameFromString(recipientName);
+                if (recipientName == "me")
+                {
+                    recipientIsPerson = true;
+                    recipientName = message.Sender.Username;
+                    recipientId = message.Sender.UserId;
+                }
+            }
+
+            var response = SendRandomResponse(message.Sender, recipientIsPerson, recipientName, recipientId, message.RoomId);
+            return Task.FromResult(new Message
+            {
+                MessageParts = response
+            });
+        }
+
+        private MessagePart[] SendRandomResponse(Person sender, bool recipientIsPerson, string recipientName,
+            string recipientId, string roomId)
+        {
+            var random = _randomnessHandler.Random.Next(7);
+            switch (random)
+            {
+                case 0:
+                    SendQueuedMessage(new Message {Text = "Kidding!", RoomId = roomId});
+                    return GetRecipientPegRemovedResponse(recipientIsPerson, recipientName, recipientId);
```

Comment:
So if I'm understanding this correctly, we first queue up the second message, and *then* return the immediate response message?

I like the fact that this means we're not waiting around inside the responder thread, but it feels sort of awkward. Is there a way to return nothing at all, and instead queue up both options such that the first happens immediately and the second happens on a delay?

Reply:
Sure there is, but I'm not sure there's enough value in it to bother changing it.

Reply:
I dunno, I sorta think of this one as the main point. Something feels *really* wrong to me about the idea of returning a response which is meant to be temporary and then later return a different response based on a timer from a helper task.